### ISSUE TITLE
lint: underef false positive in method call

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#520]
If we have a type `T` that has a method `M` which receives `*T`, we shouldn't warn on this:
```go
(*v).M() // v has type T
```
Because `(*v)` makes a copy in the original code and does not in the `v.M()`.

Reproducer:
```
$GOPATH/src/github.com/coreos/etcd/etcdserver/raft.go:450:27: underef: could simplify (*cl).Member to cl.Member
```

**Comment by Quasilyte** [comment]
Found during #513
